Megaparsec/Megaparsec.lean

```diff
@@ -44,7 +44,7 @@ instance ord2beq_ei [Ord T]: BEq (ErrorItem T) where
     | .eof, .eof => true
     | _, _ => false
 
-class Stream (S : Type) where
+structure Stream (S : Type) where
   Token : Type
```

Comment:
Hm, that curious

Reply:
But we need `class Stream` to show particular data types might be considered as a kind of stream. Do we preserve this semantics if we introduce `Stream` as a structure?

Reply:
From what I saw in the code, you needed to handle the actual instance of `Stream` many times, that's why I changed it to a structure and used parens instead of braces.
If the goal cannot be achieved with a structure then we might consider a class